modules/addons/aws/external-dns/main.tf

```diff
@@ -0,0 +1,61 @@
+resource "aws_route53_zone" "cluster" {
+  count  = var.external_dns_enabled == true ? 1 : 0
+  name   = "${var.cluster_name}.${var.dns_zone}"
+  tags   = merge(var.tags, {
+          Name = "${var.cluster_name}.${var.dns_zone}"
+  })
+}
```

Comment:
This will create the zone according to the `dns_zone` configured in the variables, but the two resources below only take into account when that zone is in the "shared zone".

Now teams have their own accounts. For example, in the MP team, we've been assigned the `mp.aws.tetrate.com` zone in the MP AWS account. With the current code, the cluster subzone that will be created there will not work.

I think we should get rid of the distinction between shared/private, and do all this stuff always, assuming that each account will already have a properly configured DNS zone, which is where we're going.